3-UsingAPIs/Week2/homework/ex1-programmerFun/index.js

```diff
@@ -17,27 +17,40 @@
 ------------------------------------------------------------------------------*/
 function requestData(url) {
```

Comment:
When writing a asynchronous function (or actually any function tbh), it's important to keep in mind not only what it returns, but **whether it could throw an error**. In this case, because you `.catch` any errors from `fetch` and don't  re-throw them, your `requestData` function will _never_ throw an error anymore.

If that is what you want, then it's OK, but I think you actually want `requestData` to throw an error, because you're wrapping it with try/catch in `main`, below. So then you should do one of these things:

- either, re-throw after catching: `.catch((error) => { console.error(...); throw error; })`
- or else (probably better), just don't catch it, and let the try/catch in `main` log the error if it happens

---

Also, remember that it's also fine to write (and good to get comfortable with the syntax):

```js
async function requestData(url) {
  // TODO return a promise using `fetch()`
  const response = await fetch(url);
  const data = await response.json();
}
```